sdk/python/tests/test_historical_retrieval.py

```diff
@@ -335,7 +336,8 @@ def test_historical_features_from_bigquery_sources(
     ) = generate_entities(start_date, infer_event_timestamp_col)
 
     # bigquery_dataset = "test_hist_retrieval_static"
-    bigquery_dataset = f"test_hist_retrieval_{int(time.time())}"
```

Comment:
I think we may still run into this problem again if we are using seconds and we are running multiple tests. For example if we vary some other dimension (that isnt the python version).

What if we did something like

Problem
```
>>> time.time()
1622217984.4427726
>>> int(time.time())
1622217987
```
Solution?
```
>>> time.time_ns()
1622218044584774799
>>> int(time.time_ns())
1622218049461962906
```

Reply:
I think that's feasible - my preference would be to use github actions' job_id or run_id cause those are also guaranteed to be unique, because ns timestamps are probably never going to collide but there's still the possibility 🤷🏽 

Would you be okay with using those environment variables (with a default value of you are running locally)?

Reply:
yeah, fair, I was concerned about that as well - I'll just do `time_ns` along with a random number and it should be good enough to avoid the underlying issue